pkg/utils/controller/run.go

```diff
@@ -58,21 +58,20 @@ func Run(ctx context.Context, logger logr.Logger, controllerName string, period
 	logger.Info("starting ...")
 	defer logger.Info("stopped")
 	var wg sync.WaitGroup
+	var workerwg wait.Group
 	defer wg.Wait()
 	defer runtime.HandleCrash()
 	metric := newControllerMetrics(logger, controllerName)
-	func() {
+	go func() {
```

Comment:
this shouldn't be a goroutine

Reply:
without including it in a go routine it is blocking the spinning up of the other controllers, after spinning up only one controller for the very first resource it goes to the blocked state and couldn't spin up the controllers for the other resources.

Reply:
yes, that's expected behaviour, controllers are started in their own go routines.

Reply:
but won't this be an issue for the TTL manager, as it will unable to sprin up the controller for particular resources?

Reply:
it will cause the faiure of the TTL kuttl tests as there will be no dedicated controller for the particular resource used in the kuttl test.

Reply:
true, but changing some code that is correct is not the right fix ;)

Reply:
so according to your opinion what should be the correct approach to tackle this.

Reply:
I could find out only this way to be specific to solve this issue.

Reply:
Running a controller should stay a blocking operation, the caller puts the call in a go routine if he wants to.

Reply:
So should I remove the func() from the go routine, simply.

Reply:
I'm going to push the fix in a minute 🤞

Reply:
> So should I remove the func() from the go routine, simply.

Not just that, the caller need to wrap the start in go routine too.